build/build.go

```diff
@@ -805,6 +800,26 @@ func BuildWithResultHandler(ctx context.Context, nodes []builder.Node, opt map[s
 		}
 	}
 
+	// local state
+	var lsg *localstate.StateGroup
+	if len(opt) > 1 {
+		lsg = &localstate.StateGroup{
+			Ref: identity.NewID(),
+		}
+	}
+	for k, opt := range opt {
```

Comment:
If this doesn't complicate it a lot then I think it can be in a goroutine.

Reply:
Moved back to first opt iteration